src/main/java/core/basesyntax/service/impl/FruitTransactionImpl.java

```diff
@@ -0,0 +1,61 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.dao.FruitShopDao;
+import core.basesyntax.dao.FruitShopDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.FruitTransaction;
+import core.basesyntax.service.strategy.OperationStrategy;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FruitTransactionImpl implements FruitTransaction {
+    private static final int INDEX_OPERATION = 0;
+    private static final int INDEX_NAME = 1;
+    private static final int START_FIND_ELEMENTS = 1;
+    private static final int INITIALIZATION_QUANTITY_IN_DB = 0;
+    private static final String SPLIT_CHARACTER = ",";
+    private static FruitShopDao fruitShopDao = new FruitShopDaoImpl();
+    private OperationStrategy operationStrategy;
+
+    public FruitTransactionImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void dateProcessing(List<String> data) {
+        validateDB(data);
```

Comment:
let's rewrite your logic, so you can remove this method from this class, you don't need to initialize fruits before processing each operation